app/models/grda_warehouse/hud/client.rb

```diff
@@ -432,8 +432,18 @@ class << self
       end
     end
 
-    scope :full_housing_release_on_file, -> do
-      where(housing_release_status: full_release_string)
+    scope :full_housing_release_on_file, ->(coc_codes: nil) do
+      # where(housing_release_status: full_release_string)
+      or_clauses = c_t[:housing_release_status].eq("#{full_release_string} for all CoCs")
```

Comment:
This is sort-of the right idea, but I think this would be something like `full_housing_release_on_file_in_coc(coc_code)`.  Then the logic will need to either save the coc_codes onto the client record when added, or join any active full releases in the right CoC.